### PR TITLE
Pytest and numerical failure diagnostics

### PR DESCRIPTION
**Pull request by newling**

### CODE REVIEW

**Comment by makslevental** [comment]
I respect the hustle

File: build_tools/ci/cpu_comparison/output_comparer.py [hunk 1]
```diff
@@ -2,65 +2,147 @@
...
+        return summary
+
+    def getSummaryString(self, max_discrepancies_to_show):
+        summary = self.getBaseString()
+        summary += self.getDiscrepenciesString(max_discrepancies_to_show)
+        summary += self.getSlicesString()
+        return summary
+
+
+def compare(npy_baseline1, npy_aie1, rtol, atol, max_discrepancies_to_show=50):
```
Comment (yzhang93) [inline build_tools/ci/cpu_comparison/output_comparer.py:129]: Why not use "npy_baseline" and "npy_aie"?

**Review by yzhang93 — APPROVED** [review]
Nice feature for debugging! Thanks!

File: build_tools/ci/cpu_comparison/output_comparer.py (hunk 1 above)
Comment (newling) [reply build_tools/ci/cpu_comparison/output_comparer.py]: I can't remember the history here, but I've dropped the '1's.

**Pull request merged**